src/main/java/com/chainstep/atlas/example/CustomPolicy.java

```diff
@@ -5,27 +5,142 @@
+                log.info("isFromLegalRegistrationNumber true");   
+            }
+            else{
+                log.info("isFromLegalRegistrationNumber false! NO access!");
+                return VerificationPolicyResult.Companion.failure();  // Fail if the legal registration number does not match
+            }
+
+            // Check if the credential has the required role
+            if(isRole(credentialNode, roleToGrantAccessString)){
+                log.info("isRole true");   
+            }
+            else{
+                log.info("isRole false! NO access!");
+                return VerificationPolicyResult.Companion.failure();  // Fail if the role is not "remoteControl"
+            }
+
+            // If all checks pass, return success
+            return VerificationPolicyResult.Companion.success();
+            
+        } catch (Exception e) {
+            log.info("error during parsing credential subject" + e);
+            return VerificationPolicyResult.Companion.failure(new Exception("parsing Error in the CustomPolicy"));
+        }
+
     }
+
+        private boolean isSignedByAtlasXAuth(JsonNode jsonNode ) {
+            // get to the legalRegistrationNumber field
+            JsonNode proofCreator = jsonNode.get("proof").get("verificationMethod");
+            log.info("proofCreator  : " + proofCreator.toString());
+            log.info("issuer  : " + jsonNode.get("issuer").asText());
+            
+            if (proofCreator.toString().contains("did:web:authority.atlas.cartrust.com:gx:did")){
+                return true;
+            }
+            else{
+                return false;
+            }
+        }
+
+    // Method to check if the credential's legal registration number matches the expected one
+    private boolean isFromLegalRegistrationNumber(JsonNode jsonNode, String legalRegistrationNumber) {
+        // Get the legal registration number from the credentialSubject field
+        JsonNode legalRegistrationNumberNode = jsonNode
+            .get("credentialSubject")
+            .get("legalRegistrationNumber");
+
+            log.info("legalRegistrationNumber  : " + legalRegistrationNumberNode.asText());
+            
+            if (legalRegistrationNumberNode.asText().equals(legalRegistrationNumber)){
+                return true;
+            }
+            else{
+                return false;
+            }
+        }
+
+        private boolean isRole(JsonNode jsonNode, String role) {
+            // get to the legalRegistrationNumber field
+    
+            JsonNode roleNumberNode = jsonNode
+            .get("credentialSubject")
+            .get("role");
+    
+        log.info("role  : " + roleNumberNode.asText());
+        
+        // Check if the role matches
+        if (roleNumberNode.asText().equals(role)){
+            return true;
+        }
+        else{
+            return false;
+        }
+    }
```

Comment:
There is a formatting issue on this file, please use the plugin "Language Support for Java(TM) by Red Hat" and reformat the file.

Reply:
done